stargz/stargz.go

```diff
@@ -729,9 +732,18 @@ func (w *Writer) AppendTar(r io.Reader) error {
 			return fmt.Errorf("unsupported input tar entry %q", h.Typeflag)
 		}
 
+		// We need to keep a reference to the TOC entry for regular files, so that we
+		// can fill the digest later.
+		var regFileEntry *TOCEntry
+		if h.Typeflag == tar.TypeReg {
+			regFileEntry = ent
+		}
+		payloadDigest := sha256.New()
```

Comment:
No need to make this for directories, etc, but I guess it also doesn't matter. AppendTar isn't a hot path.

Reply:
I've changed it to be done only for TypeReg